### ISSUE TITLE
Zeebe 8.3.2 JournalException OutOfDiskSpace: Not enough space to allocate a new journal segment

### ISSUE THREAD

**Issue by nicpuppa** [camunda/zeebe#15453]
This error happens 945 times from the 24/11 ~ 5pm to 28/11 ~ 5pm. 

This error seems happen everytime on the same node `zeebe-0` during a restart. 

Error log:

```
io.camunda.zeebe.journal.JournalException$OutOfDiskSpace: Not enough space to allocate a new journal segment. Required: 2147483648, Available: 1003868160

at io.camunda.zeebe.journal.file.SegmentLoader.checkDiskSpace ( io/camunda.zeebe.journal.file/SegmentLoader.java:200 )
at io.camunda.zeebe.journal.file.SegmentLoader.mapNewSegment ( io/camunda.zeebe.journal.file/SegmentLoader.java:174 )
at io.camunda.zeebe.journal.file.SegmentLoader.createSegment ( io/camunda.zeebe.journal.file/SegmentLoader.java:55 )
at io.camunda.zeebe.journal.file.SegmentsManager.createSegment ( io/camunda.zeebe.journal.file/SegmentsManager.java:342 )
at io.camunda.zeebe.journal.file.SegmentsManager.open ( io/camunda.zeebe.journal.file/SegmentsManager.java:302 )
at io.camunda.zeebe.journal.file.SegmentedJournal.<init> ( io/camunda.zeebe.journal.file/SegmentedJournal.java:55 )
at io.camunda.zeebe.journal.file.SegmentedJournalBuilder.build ( io/camunda.zeebe.journal.file/SegmentedJournalBuilder.java:181 )
at io.atomix.raft.storage.log.RaftLogBuilder.build ( io/atomix.raft.storage.log/RaftLogBuilder.java:151 )
at io.atomix.raft.storage.RaftStorage.openLog ( io/atomix.raft.storage/RaftStorage.java:197 )
at io.atomix.raft.impl.RaftContext.<init> ( io/atomix.raft.impl/RaftContext.java:207 )
at io.atomix.raft.impl.DefaultRaftServer$Builder.build ( io/atomix.raft.impl/DefaultRaftServer.java:242 )
at io.atomix.raft.impl.DefaultRaftServer$Builder.build ( io/atomix.raft.impl/DefaultRaftServer.java:216 )
at io.atomix.raft.partition.impl.RaftPartitionServer.buildServer ( io/atomix.raft.partition.impl/RaftPartitionServer.java:159 )
at io.atomix.raft.partition.impl.RaftPartitionServer.<init> ( io/atomix.raft.partition.impl/RaftPartitionServer.java:91 )
at io.atomix.raft.partition.RaftPartition.createServer ( io/atomix.raft.partition/RaftPartition.java:127 )
at io.atomix.raft.partition.RaftPartition.initServer ( io/atomix.raft.partition/RaftPartition.java:110 )
at io.atomix.raft.partition.RaftPartition.bootstrap ( io/atomix.raft.partition/RaftPartition.java:94 )
at io.camunda.zeebe.broker.partitioning.startup.steps.RaftBootstrapStep.startup ( io/camunda.zeebe.broker.partitioning.startup.steps/RaftBootstrapStep.java:31 )
at io.camunda.zeebe.broker.partitioning.startup.steps.RaftBootstrapStep.startup ( io/camunda.zeebe.broker.partitioning.startup.steps/RaftBootstrapStep.java:14 )
at io.camunda.zeebe.scheduler.startup.StartupProcess.proceedWithStartupSynchronized ( io/camunda.zeebe.scheduler.startup/StartupProcess.java:162 )
at io.camunda.zeebe.scheduler.startup.StartupProcess.lambda$proceedWithStartupSynchronized$3 ( io/camunda.zeebe.scheduler.startup/StartupProcess.java:170 )
at io.camunda.zeebe.scheduler.future.FutureContinuationRunnable.run ( io/camunda.zeebe.scheduler.future/FutureContinuationRunnable.java:28 )
at io.camunda.zeebe.scheduler.ActorJob.invoke ( io/camunda.zeebe.scheduler/ActorJob.java:96 )
at io.camunda.zeebe.scheduler.ActorJob.execute ( io/camunda.zeebe.scheduler/ActorJob.java:47 )
at io.camunda.zeebe.scheduler.ActorTask.execute ( io/camunda.zeebe.scheduler/ActorTask.java:120 )
at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask ( io/camunda.zeebe.scheduler/ActorThread.java:130 )
at io.camunda.zeebe.scheduler.ActorThread.doWork ( io/camunda.zeebe.scheduler/ActorThread.java:108 )
at io.camunda.zeebe.scheduler.ActorThread.run ( io/camunda.zeebe.scheduler/ActorThread.java:227 )
```

Just before this error there is another interesting warning related to the broker

```Out of disk space. Current available 1003868160 bytes. Minimum needed 3221225472 bytes.```

The cluster seems not be more active since 28/11 at around 5pm. The cluster was completely unhealthy and in a restarting loop for 4 days

More logs [here](https://drive.google.com/drive/folders/1I7IiI_cCFCr7varv4LA8ECr6jUO6HNVT)
Link to the [error reporting tool](https://console.cloud.google.com/errors/detail/CJCNuP-4tqeM0QE;service=zeebe;time=P7D?project=camunda-saas-dev)

**Comment by Zelldon** [comment]
@nicpuppa I guess you have seen this via the error reporting tool. Please link to the actual error group, and do some more root causing.

When does it happen? Where there any other events before?
Did it recover? Was it only on one node? What is the impact for the cluster, etc.

**Comment by npepinpe** [comment]
Could you please look to see if:

- There were any other errors in this cluster
- Did we go out of disk space because of external factors, e.g. Elastic was unavailable, we couldn't export, etc.?

And if you have no time, please delegate to someone via Slack or just ping me here.

**Comment by nicpuppa** [comment]
Hey @npepinpe. I didn't saw any other errors in the cluster, and the reason of the out of disk space is not clear. As I mentioned above, the error occurred after 12 second the first log (starting the standalone broker). The only warning visible is 

```"No backup store is configured. Backups will not be taken"```

**Comment by npepinpe** [comment]
Was this a cluster restart? In which case the issue might be from before the start. If it was a complete fresh cluster start, it might be either a misconfiguration (e.g. bad disk watermarks configured) or some infra issue (disks were too small?)

**Comment by nicpuppa** [comment]
No it was not a restart, it was a fresh cluster start

**Comment by npepinpe** [comment]
OK, looks like an environmental error, that means from 1003900928 bytes the disk, _on a fresh start_, only had 1GB, which is way too low. Can you ping SREs to look into that? Or maybe the controller team was doing some tests?

**Comment by npepinpe** [comment]
Sounds good

**Comment by korthout** [comment]
This occurred [again](https://console.cloud.google.com/errors/detail/CJCNuP-4tqeM0QE;service=zeebe;time=P7D?project=camunda-cloud-240911). Details:
- error: `io.camunda.zeebe.journal.JournalException$OutOfDiskSpace: Not enough space to allocate a new journal segment. Required: 1073741824, Available: 1003864064`
- context: Unable to boot
- namespace: `zbcluster-controller-e2e-small-connection-zeebe`
- version: `8.5.0-alpha2`
- env: `production-worker-1`

I'll check with Controller team again.

<img width="1343" alt="Screenshot 2024-03-18 at 17 23 29" src="https://github.com/camunda/zeebe/assets/3511026/d8068983-f7b4-48c9-adb7-8efb36687421">
